cpp/ql/src/semmle/code/cpp/exprs/Expr.qll

```diff
@@ -68,6 +68,16 @@ class Expr extends StmtParent, @expr {
    */
   Type getUnderlyingType() { result = this.getType().getUnderlyingType() }
 
+  /**
+   * Gets the type of this expression after specifiers have been deeply
+   * stripped and typedefs have been resolved.
+   *
+   * In most cases, this predicate will be the same as getType().  It will
+   * only differ when the result of getType() is a TypedefType, in which
+   * case this predicate will (possibly recursively) resolve the typedef.
```

Comment:
The second paragraph seems to be identical to the text in `getUnderlyingType`, but the predicates do different things. Please either adjust or remove the text. If you adjust it, add the same text to all the other `getUnderlyingType` predicates. It might be helpful for users to say that a _specifier_ means `const`, `volatile`, or similar.